### PR TITLE
Clean up a few plain `e.printStackTrace()` calls

### PR DESCRIPTION
**Pull request by Vlatombe**
And use logger with appropriate severity instead.

### Proposed changelog entries

- N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/hudson/model/ViewJob.java
Comment (Vlatombe) [inline core/src/main/java/hudson/model/ViewJob.java:244]: ⚠️

File: core/src/main/java/hudson/model/AbstractItem.java
Comment (jglick) [inline core/src/main/java/hudson/model/AbstractItem.java:437]:
```suggestion
                            LOGGER.log(Level.WARNING, "Ignoring IOException while deleting", e);
```
though this code seems pretty fragile to begin with.

File: core/src/main/java/hudson/model/ViewJob.java
Comment (jglick) [reply core/src/main/java/hudson/model/ViewJob.java:244]: Yeah. This class should just be moved to https://github.com/jenkinsci/external-monitor-job-plugin/blob/9fc4fed70ffac8b8c52288359c7854bffcb498c4/src/main/java/hudson/model/ExternalJob.java#L46 anyway.

File: core/src/main/java/jenkins/model/Jenkins.java
Comment (jglick) [inline core/src/main/java/jenkins/model/Jenkins.java:5635]:
```suggestion
            LOGGER.log(Level.WARNING, e, () -> "Failed to load jenkins-version.properties");
```

Why would this be OK?

File: core/src/main/java/hudson/model/ViewJob.java
Comment (jglick) [inline core/src/main/java/hudson/model/ViewJob.java:254]:
```suggestion
                        LOGGER.log(Level.WARNING, e, () -> "Failed to reload job " + finalJobName);
                    } else {
                        LOGGER.log(Level.WARNING, e, () -> "Failed to obtain next job in the reload queue");
```
probably (I doubt these are expected conditions)

**Review by jglick — APPROVED** [review]

File: core/src/main/java/jenkins/model/Jenkins.java
Comment (Vlatombe) [reply core/src/main/java/jenkins/model/Jenkins.java]: Should be fine, as it should only happen in case of a real packaging issue.

**Review by jglick — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**
